pkg/configuration/configuration.go

```diff
@@ -0,0 +1,94 @@
+// Package configuration is in charge of the validation and extraction of all
+// the configuration details from a configuration file or environment variables.
+package configuration
+
+import (
+	"os"
+	"strings"
+
+	errs "github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+// prefixes
+const (
+	// MemberEnvPrefix will be used for member environment variable name prefixing.
+	MemberEnvPrefix = "MEMBER_OPERATOR"
+)
+
+// Configuration constants
+const (
+	// ToolchainConfigMapName specifies a name of a ConfigMap that keeps toolchain configuration
+	ToolchainConfigMapName = "toolchain-saas-config"
+
+	// IdentityProvider specifies an identity provider (IdP) for newly created users
+	IdentityProvider = "identity.provider"
+
+	// DefaultIdentityProvider the default value used for the identity provider (IdP) for newly created users
+	DefaultIdentityProvider = "rhd"
+)
+
+// Registry encapsulates the Viper configuration registry which stores the
+// configuration data in-memory.
+type Registry struct {
+	member *viper.Viper
+}
+
+// CreateEmptyRegistry creates an initial, empty registry.
+func CreateEmptyRegistry() *Registry {
```

Comment:
this func does not need to be public AFAICS

Reply:
Appears so, good catch! 👍